src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -672,10 +672,16 @@ protected function dumpContainer(ConfigCache $cache, ContainerBuilder $container
      *
      * @param ContainerInterface $container The service container
      *
+     * @throws \DomainException If the container is not of type ContainerBuilder
+     *
      * @return DelegatingLoader The loader
      */
     protected function getContainerLoader(ContainerInterface $container)
     {
+        if (!$container instanceof ContainerBuilder) {
+            throw new \DomainException(sprintf('Container must be an instance of Symfony\Component\DependencyInjection\ContainerBuilder, got %s', get_class($container)));
```

Comment:
Just my opinion but I would prefer see `The first argument of ...\Kernel::getContainerLoader() must be an instance of...` for faster understanding